### PR TITLE
Use Axes.tick_params/Axis.set_tick_params more

### PR DESCRIPTION
**Pull request by QuLogic**
## PR Summary

This PR replaces a lot of manual loops over ticks/tick labels with calls to `Axes.tick_params`/`Axis.set_tick_params`. These methods work on all ticks at the same time and using them will allow for some optimization in a later PR.

Ping @choldgraf since most of these are examples and to make sure I didn't botch anything there.

## PR Checklist

- [x] Has Pytest style unit tests (existing ones, anyway)
- [x] Code is PEP 8 compliant 
- [N/A] New features are documented, with examples if plot related
- [x] Documentation is sphinx and numpydoc compliant
- [N/A] Added an entry to doc/users/whats_new.rst if major new feature
- [N/A] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: examples/pylab_examples/axes_props.py [hunk 1]
```diff
@@ -11,22 +11,11 @@
...
 fig, ax = plt.subplots()
 ax.plot(t, s)
-ax.grid(True)
-
-ticklines = ax.get_xticklines() + ax.get_yticklines()
-gridlines = ax.get_xgridlines() + ax.get_ygridlines()
-ticklabels = ax.get_xticklabels() + ax.get_yticklabels()
-
-for line in ticklines:
-    line.set_linewidth(3)
```
Comment (QuLogic) [inline examples/pylab_examples/axes_props.py]: As noted in the documentation above, this was actually wrong, since ticks use markers.

**Comment by QuLogic** [comment]
@tacaswell these are the sorts of things I was talking about on gitter.

File: doc/faq/howto_faq.rst [hunk 2]
```diff
@@ -241,17 +241,29 @@ over so that the tick labels fit in the figure:
...
+
+    ax.tick_params(width=10)
+
+    plt.show()
+
+For more control of tick properties that are not provided by the above methods,
+it is important to know that in Matplotlib, the ticks are *markers*.  All
+:class:`~matplotlib.lines.Line2D` objects support a line (solid, dashed, etc)
+and a marker (circle, square, tick).  The tick width is controlled by the
+"markeredgewidth" property::
```
Comment (choldgraf) [inline doc/faq/howto_faq.rst]: why not `` ?

File: examples/api/logos2.py [hunk 3]
```diff
@@ -71,14 +71,10 @@ def add_polar_bar():
...
 
-    for label in ax.get_xticklabels() + ax.get_yticklabels():
-        label.set_visible(False)
-
-    for line in ax.get_ygridlines() + ax.get_xgridlines():
-        line.set_lw(0.8)
-        line.set_alpha(0.9)
-        line.set_ls('-')
-        line.set_color('0.5')
+    ax.tick_params(labelbottom=False, labeltop=False,
```
Comment (choldgraf) [inline examples/api/logos2.py]: I had no idea this was possible...cool!

**Comment by choldgraf** [comment]
LGTM. Nice job on reducing the total number of lines in matplotlib 👍  :)

File: doc/faq/howto_faq.rst (hunk 2 above)
Comment (QuLogic) [inline doc/faq/howto_faq.rst]: Probably we just didn't use it much when this was first written.

Comment (QuLogic) [inline doc/faq/howto_faq.rst]: Also, I just noticed the below example doesn't actually use `markeredgewidth`...

**Comment by QuLogic** [comment]
In cleaning up the doc things you mentioned, I noticed that I had not updated the tutorials. However, the tutorials actually hit one of the shortcomings of `ax.tick_params`: it works on both tick markers and tick labels, so it only supports a limited set of options for each. Thus the [tutorial that sets horizontal alignment](https://github.com/matplotlib/matplotlib/blob/e169aff9a338efcb03f016016c369d9b46f63858/tutorials/01_introductory/lifecycle.py#L159) is not possible directly with `ax.tick_params`. We probably need to come up with some new API to encompass these sorts of things.

Ironically, when the tutorials explain why to use `ax.get_tick*`, it's actually a reason to avoid using it (at least for any interactive plots.)

**Comment by QuLogic** [comment]
Fixed the doc comment, and I'm going to leave the tutorial stuff until after we've finalized the other API changes.

File: lib/matplotlib/projections/polar.py [hunk 4]
```diff
@@ -561,8 +560,7 @@ def set_rgrids(self, radii, labels=None, angle=None, fmt=None,
         if angle is None:
             angle = self.get_rlabel_position()
         self.set_rlabel_position(angle)
-        for t in self.yaxis.get_ticklabels():
-            t.update(kwargs)
+        self.yaxis.set_tick_params(which='major', **kwargs)
```
Comment (anntzer) [inline lib/matplotlib/projections/polar.py]: Is this really equivalent (it probably should, but we all know how consistent the API is)?  set_tick_params ultimately forwards the arguments to Tick._apply_params, which is different from Tick.update (i.e. Artist.update).

Comment (QuLogic) [inline lib/matplotlib/projections/polar.py]: Unfortunately, you're right to be suspicious. There's a loop through the input to check that the keys are valid for ticks, and doesn't necessarily allow all possible `Line2D` arguments.

File: lib/matplotlib/figure.py [hunk 5]
```diff
@@ -1173,12 +1173,14 @@ def subplots(self, nrows=1, ncols=1, sharex=False, sharey=False,
         if sharex in ["col", "all"]:
             # turn off all but the bottom row
             for ax in axarr[:-1, :].flat:
-                ax.xaxis.set_tick_params(labelbottom=False)
+                ax.xaxis.set_tick_params(which='both',
```
Comment (tacaswell) [inline lib/matplotlib/figure.py]: The you had it might be better.

Comment (QuLogic) [inline lib/matplotlib/figure.py]: Well, we need `which='both'` or the test fails, unless you mean `bottom` instead of `1`?

Comment (tacaswell) [inline lib/matplotlib/figure.py]: When I wrote this I went back and forth on if removing _all_ of the labels vs just the bottom and left was overkill.

'labelbottom' and 'labelleft' are certainly more readable!

**Pull request merged**